### PR TITLE
:bug: Don't use NodeLocalDNS addon

### PR DESCRIPTION
**Pull request by cardil**
# Changes

- :bug: Don't use NodeLocalDNS addon

/kind bug

Fixes partially https://github.com/knative-sandbox/kn-plugin-event/issues/242

### CODE REVIEW

File: e2e-tests.sh [hunk 1]
```diff
@@ -178,7 +178,7 @@ function initialize() {
   done
 
   if [[ "${CLOUD_PROVIDER}" == "gke" ]]; then
-      custom_flags+=("--addons=NodeLocalDNS")
+      custom_flags+=("--command-group=") # Use production GKE commands, instead of beta.
```
Comment (cardil) [inline e2e-tests.sh:181]: When https://github.com/knative/test-infra/pull/3623 is merged this might be removed.

Comment (cardil) [reply e2e-tests.sh]: Removed in 5af6017e0b2bdb5653977c71f508e3e82940ad36

**Comment by cardil** [comment]
Here's a justification of this change: https://github.com/knative-sandbox/kn-plugin-event/issues/242#issuecomment-1322745383

Here's a related guide from Google: https://cloud.google.com/kubernetes-engine/docs/how-to/nodelocal-dns-cache#known_issues

**Comment by kvmware** [comment]
Sorry misclicked close instead of comment for the `/lgtm`, my bad :/

**Comment by cardil** [comment]
It turns out the _NodeLocalDNS_ wasn't the culprit... I did run 9 runs in a row using production GKE with _NodeLocalDNS_ enabled. No errors. Same on https://github.com/knative-sandbox/kn-plugin-event/pull/250 already 3 runs in a row.

#258 will revert this (without the kubetest2 convenience install).

**Pull request merged**